src/mlui/enums/Adabound.py

```diff
@@ -0,0 +1,225 @@
+
+
+
+import tensorflow.compat.v2 as tf
+from keras import backend
+from keras.optimizers import Optimizer
+
+class AdaBound(Optimizer):
+    """
+    Adabound optimizer implementation 
+    # References
+        - [Adaptive Gradient Methods with Dynamic Bound of Learning Rate]
+          (https://openreview.net/forum?id=Bkg3g2R9FX)
+
+
+
+    Parameters
+    -----------
+        learning_rate : float 
+            Adam learning rate (default: 1e-3)
+
+        base_lr: float >= 0.
+             Used for loading the optimizer. Do not set the argument manually.    
+
+        
+        final_lr : float
+                  final (SGD) learning rate (default: 0.1)
+
+        beta1,beta2 : float between [0,1] 
+            coefficients used for computingrunning averages of gradient and its square (default: (0.9, 0.999))
+          
+        gamma : float
+                convergence speed of the bound functions (default: 1e-3)
+
+        eps : float
+            term added to the denominator to improve numerical stability (default: 1e-8)
+
+        weight_decay : float 
+             weight decay (L2 penalty) (default: 0)
+
+        amsbound :boolean
+                whether to use the AMSBound variant of this algorithm   
+    """
+    def __init__(
+        self,
+        learning_rate=0.001,
+        base_lr = None,
+        final_lr=0.1,
+        beta_1=0.9,
+        beta_2=0.999,
+        gamma = 1e-3,
+        epsilon=1e-7,
+        amsgrad=False,
+        weight_decay=None,
+        clipnorm=None,
+        clipvalue=None,
+        global_clipnorm=None,
+        use_ema=False,
+        ema_momentum=0.99,
+        ema_overwrite_frequency=None,
+        jit_compile=True,
+        name="AdaBound",
+        **kwargs
+    ):
+
+        super().__init__(
+            name=name,
+            weight_decay=weight_decay,
+            clipnorm=clipnorm,
+            clipvalue=clipvalue,
+            global_clipnorm=global_clipnorm,
+            use_ema=use_ema,
+            ema_momentum=ema_momentum,
+            ema_overwrite_frequency=ema_overwrite_frequency,
+            jit_compile=jit_compile,
+            **kwargs
+        )
+        
+        if base_lr is None:
+            if isinstance(learning_rate, tf.keras.optimizers.schedules.LearningRateSchedule):
+                base_lr = learning_rate(0)
+            else:
+                base_lr = learning_rate
+        self._learning_rate = self._build_learning_rate(learning_rate)
+        self.base_lr = base_lr
+        self.fianl_lr = final_lr
+        self.beta_1 = beta_1
+        self.beta_2 = beta_2
+        self.gamma = gamma
+        self.epsilon = epsilon
+        self.amsgrad = amsgrad
+
+
+
+    def build(self, var_list):
+         """ AdaBound optimizer has 3 types of variables: momentums, velocities and
+        velocity_hat (only set when amsgrad is applied),
+
+        parameters
+        ----------
+            var_list: list of model variables to build AdaBound variables on.
```

Comment:
There should be a space before the semicolon `:`. The description should be placed bellow the type annotation.

Reply:
solved.  because the Type of _var_list_ is absent the colon has been omitted.